File: javasript/check-array-formation-through-concatenation.js
/**
 * @param {number[]} arr
 * @param {number[][]} pieces
 * @return {boolean}
 */
var canFormArray = function(arr, pieces) {
      const n = arr.length;
        for(let e=0;e<arr.length;++e){
          
            let state= false;
            outer:
            for(let i = 0;i<pieces.length;++i){
               
                if(pieces[i][0]==arr[e] && pieces[i].length<=n-e){
                    let idx=e;
                   for(let element of pieces[i]){
                       if(element != arr[idx]) return false;
                       ++idx;
                   }
                    
                    state = true;
                    e=idx-1;
                    continue;
                }
            }
            if(!state)return false;
            
            
           
        }
        
        return true;
        
};

File: java/check-array-formation-through-concatenation.java
class Solution {
    public boolean canFormArray(int[] arr, int[][] pieces) {
        
        
        int n = arr.length;
        for(int e=0;e<arr.length;++e){
          
            boolean state= false;
            outer:
            for(int i = 0;i<pieces.length;++i){
               
                if(pieces[i][0]==arr[e] && pieces[i].length<=n-e){
                    int idx=e;
                   for(int element : pieces[i]){
                       if(element != arr[idx]) return false;
                       ++idx;
                   }
                    
                    state = true;
                    e=idx-1;
                    continue;
                }
            }
            if(!state)return false;
            
            
           
        }
        
        return true;
        
    }
}